### ISSUE TITLE
mesh bp: add support for pyramid and wedge elements

### ISSUE THREAD

**Issue by cyrush** [LLNL/conduit#132]

**Comment by rubenscholfield** [comment]
I think i ve got my own server with bits from telekom, they are secued layed down for me and are in work at mellowads😉. Cloud is ok, bowser else. But server, hä??? I do systemwork from annahmestelle😁 
and else i try to figure something out with this konto ????????, myne konto and support......
but first systemeinstellungen to stabilise, 😉
See you
rubenscholfield@gmail.com

**Comment by rubenscholfield** [comment]
😁

**Comment by rubenscholfield** [comment]
:-|

**Comment by rubenscholfield** [comment]
😉

**Comment by cyrush** [comment]
Adding the cases to blueprint verify is quite easy, but this also needs to include creating wedge and pyramid versions of braid mesh examples.

**Comment by FreddieWitherden** [comment]
Is this still on track for 0.9.0?  I ask, as the lack of support for square-based pyramids and triangular prisms is one of the blockers preventing PyFR from using Conduit.

**Comment by cyrush** [comment]
Yes, thanks for asking. Knowing this is a blocker we can address. 

Q:  Is this in the context - do the meshes have heterogeneous element types -- you want hexs, tets, and wedges all in one topology?

Currently, heterogeneous (or mixed element types) are only supported via arbitrary polygonal / polyhedral descriptions. 

But we want to provide a simpler path for the finite element zoo types that conduit knows about.

Another caveat to wedge and pyramid support -- initial support will allow them as valid cases in the mesh blueprint - but they may not be fulled supported the mesh blueprint transforms (things like generate_sides, etc) until later.

**Comment by FreddieWitherden** [comment]
We do work (almost entirely) with mixed grids.  For the most part we are fine with separate topologies, however.  PyFR internally has separate solution arrays for the various element types we support and so this is not much of an issue.  The only exception is with pyramids where our subdivision algorithm (we're a high-order code) breaks pyramids down into smaller pyramids and tetrahedra.  Here, if needed, we can do an extra pass to group the tets together with all of the other tets in our domain.

**Comment by cyrush** [comment]
@JustinPrivitera another Conduit task we discussed.

**Comment by cyrush** [comment]
@FreddieWitherden FYI: conduit develop now support pyramids and wedges.

**Comment by FreddieWitherden** [comment]
> @FreddieWitherden FYI: conduit develop now support pyramids and wedges.

Great!  We are close to mainlining our Ascent plugin:

https://github.com/PyFR/PyFR/pull/307

and will be sure to take advantage of this support!
